modules/storages/app/common/storages/peripherals/connection_validators/nextcloud/ampf_connection_validator.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module ConnectionValidators
+      module Nextcloud
+        class AmpfConnectionValidator < BaseValidator
+          using ServiceResultRefinements
+
+          private
+
+          def validate
+            register_checks(
+              :userless_access, :group_folder_presence, :files_request, :group_folder_contents
+            )
+
+            userless_access_denied
+            group_folder_not_found
+            files_request_failed_with_unknown_error
+            with_unexpected_content
+          end
+
+          def userless_access_denied
+            if files.result == :unauthorized
+              fail_check(:userless_access, message(:userless_access_denied))
+            else
+              pass_check(:userless_access)
+            end
+          end
+
+          def group_folder_not_found
+            if files.result == :not_found
+              fail_check(:group_folder_presence, message(:group_folder_not_found))
+            else
+              pass_check(:group_folder_presence)
+            end
+          end
+
+          def files_request_failed_with_unknown_error
+            if files.result == :error
+              error "Connection validation failed with unknown error:\n\t" \
+                    "storage: ##{@storage.id} #{@storage.name}\n\t" \
```

Comment:
:green_circle: WDYT about moving the `\t` to the beginning of the following line? That would make it clearer which lines are indented.

Reply:
Oh, I just copied over the previous line. Good catch.